web/__init__.py

```diff
@@ -1,23 +1,43 @@
+import os
+import re
 import json
+import tempfile
 import logging as log
 
-from flask import Flask, request, render_template, Response
+from secrets import token_urlsafe as token
+
+from flask import Flask, flash, request, redirect, url_for, render_template, Response, session
 from flask_sock import Sock
+from werkzeug.utils import secure_filename
+
+import libflagship.httpapi
+import libflagship.logincache
 
 from libflagship.pppp import P2PSubCmdType, FileTransfer
 from libflagship.ppppapi import FileUploadInfo, PPPPError
 
 
 from web.lib.service import ServiceManager
 
+from user_agents import parse
+
 import cli.util
+import cli.config
+
+
+def dict_match(dict, match):
+    for key in dict:
+        if re.match(rf"^{key}.*", match):
```

Comment:
`re.match` matches a substring from the start of the string, so this line is the same as:

```python
if re.match(key, match): ...
```


Also, we only match against non-regex strings, so this test becomes:

```python
if match.startswith(key): ...
```

At that point, this function becomes very light :)

I can see it's used to determine a platform-specific hint about file locations, for now.

How about we make a function that return a standardized platform name:

```python
def user_agent_platform(...):
   if ...:
     return "linux"
   ...
     return "windows"
   ...
     return "macos"
```

Let's put it in `web/utils.py`. Then, the result from this can simply be used in a dictionary lookup.

Reply:
Sorry, I meant `web/platform.py`, like I referenced in a later comment. But same idea :)